src/efs/filesystem.py

```diff
@@ -0,0 +1,122 @@
+"""The file system abstraction."""
+import urllib.parse
+
+from autorepr import autorepr
+from flask import current_app
+
+from .efosfs import EFOSFS
+from .efs3 import EFS3
+
+
+class EFS:
```

Comment:
Now that we are messing with this untouched part of the code... why not a better name like `EatFirstFS` or `EatFirstFileSystem`?

`EFS` and `EFS3` look very similar, and they should not be.
`EFS3` looks like the version 3 of `EFS`.